bdk_rpc/src/lib.rs

```diff
@@ -0,0 +1,377 @@
+/// Initiate a RPC Client for a given configuration and a Tracker.
+pub fn init_client_for_tracker<K: Debug + Clone + Ord>(
+    config: &RpcConfig,
+    tracker: &KeychainTracker<K>,
+) -> Result<Client, RpcError> {
+    let wallet_name = wallet_name_from_tracker(tracker);
+    let wallet_url = format!("{}/wallet/{}", config.url, wallet_name);
+    let client = Client::new(wallet_url.as_str(), config.auth.clone().into())?;
+    let rpc_version = client.version()?;
+    debug!("rpc connection established. Core version : {}", rpc_version);
+    info!("connected to '{}' with auth: {:?}", wallet_url, config.auth);
+
+    if client.list_wallets()?.contains(&wallet_name) {
+        info!("wallet already loaded: {}", wallet_name);
+    } else if list_wallet_dir(&client)?.contains(&wallet_name) {
+        debug!("wallet wasn't loaded. Loading wallet : {}", wallet_name);
+        client.load_wallet(&wallet_name)?;
+        info!("wallet loaded: {}", wallet_name);
+    } else {
+        // pre-0.21 use legacy wallets
+        if rpc_version < 210_000 {
+            client.create_wallet(&wallet_name, Some(true), None, None, None)?;
+        } else {
+            // TODO: move back to api call when https://github.com/rust-bitcoin/rust-bitcoincore-rpc/issues/225 is closed
+            let args = [
+                Value::String(wallet_name.clone()),
+                Value::Bool(true),  // disable_private_keys
+                Value::Bool(false), //blank
+                Value::Null,        // passphrase
+                Value::Bool(false), // avoid reuse
+                Value::Bool(false), // descriptor
+            ];
+            let _: Value = client.call("createwallet", &args)?;
+        }
+
+        info!("wallet created: {}", wallet_name);
+    }
+
+    let blockchain_info = client.get_blockchain_info()?;
+    let network = match blockchain_info.chain.as_str() {
+        "main" => Network::Bitcoin,
+        "test" => Network::Testnet,
+        "regtest" => Network::Regtest,
+        "signet" => Network::Signet,
+        _ => {
+            return Err(RpcError::Generic(
+                "Invalid network in blockchain info".to_string(),
+            ))
+        }
+    };
+    if network != config.network {
+        return Err(RpcError::InvalidNetwork {
+            requested: config.network,
+            found: network,
+        });
+    }
+
+    Ok(client)
+}
+
+/// Sync the Tracker, Sparsechain and Client with each other.
+pub fn rpc_sync<K: Debug + Clone + Ord>(
+    client: &Client,
+    tracker: &mut KeychainTracker<K>,
+    sparse_chain: &mut SparseChain,
+) -> Result<(), RpcError> {
+    let start = std::time::Instant::now();
+
+    // Import spks from Tracker to Client
+    // Can have spks that are already imported
+    let spk_map = &*tracker.script_pubkeys();
+    let _labelled_scripts: Vec<(Option<String>, Script)> = spk_map
+        .iter()
+        .map(|((keychain, _), script)| {
+            let label = Some(format!("{:?}", keychain));
```

Comment:
I think this is a nice idea. It should use `{}` rather than debug though. Require that `K` is `Display`.
(but these should be passed in rather than generated here).